pkg/kf/commands/apps/push.go

```diff
@@ -62,58 +63,75 @@ func NewPushCommand(p *config.KfParams, pusher kf.Pusher, b SrcImageBuilder) *co
   kf push myapp --container-registry gcr.io/myproject --buildpack my.special.buildpack # Discover via kf buildpacks
   kf push myapp --container-registry gcr.io/myproject --env FOO=bar --env BAZ=foo
   `,
-		Args: cobra.ExactArgs(1),
+		Args: cobra.MaximumNArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
 			if containerRegistry == "" {
 				return errors.New("container-registry is required")
 			}
 			cmd.SilenceUsage = true
 
-			// Cobra ensures we are only called with a single argument.
-			appName := args[0]
-			var imageName string
-
-			switch {
-			case sourceImage != "":
-				imageName = sourceImage
-			default:
-				// As no source image is provided, we'll have to upload one.
-				if path != "" {
-					// Use the given path, but ensure it is an absolute path.
-					// Kontext has to have a absolute path.
-					var err error
-					path, err = filepath.Abs(path)
-					if err != nil {
-						return err
-					}
-				} else {
-					// No path set, use the current working directory.
-					cwd, err := os.Getwd()
-					if err != nil {
+			appName := ""
+			if len(args) > 0 {
+				appName = args[0]
+			}
+
+			// Kontext has to have a absolute path.
+			var err error
+			path, err = filepath.Abs(path)
+			if err != nil {
+				return err
+			}
+
+			pushManifest, err := manifest.NewFromFileOrDefault(manifestFile, appName)
```

Comment:
Can we restructure this a little? Its a bit misleading giving the flag a default value of `manifest.yml` and that `NewFromFileOrDefault()` doesn't return an error if the manifest path doesn't exist. (It took me a few rounds to get how it all worked)

This also implies if a user does `kf push some-app --manifest /path/does/not/exist`, it will still try to deploy it.

Reply:
I refactored this into two methods: `CheckForManifest` which returns a manifest if it exists or `nil` otherwise and `New(appname)` which creates a default manifest for a single app.

I'm thinking of further splitting up `CheckForManifest` s.t. it returns a string pointer which points to the manifest file name.

Reply:
I like that quite a bit better!